cstore_fdw--1.5.sql

```diff
@@ -58,3 +60,37 @@ CREATE EVENT TRIGGER cstore_drop_event
     ON SQL_DROP
     EXECUTE PROCEDURE cstore_drop_trigger();
 
+CREATE TABLE public.pg_cstore_tables (logicalrelid regclass PRIMARY KEY);
```

Comment:
High level suggestion:
 (1) Add data_size and block_row_count to pg_cstore_tables,
 (2) StripeHeader = StripeFooter + StripeMetadata (with unnecessary fields removed). Put this at beginning of stripe. No need to store StripeMetadata at StripeFooter.
 (3) With 1 and 2, we don't need TableFooter anymore.

For reads, we know the start position of next stripe after reading a stripe's header. We also know how much data is visible before current scan started. I guess these are sufficient to scan through the whole table.

For writes, by looking at data_size we know where to start appending.

I think this simplifies the layout, and removes the need for some of the functions that are not very easy to understand.

What do you think?